src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php

```diff
@@ -155,6 +155,12 @@ private function compileRoutes(RouteCollection $routes, $supportsRedirections)
             }
         }
 
+        if ('' === $code) {
+            $code .= "        if ('/' === \$pathinfo) {\n";
+            $code .= "            throw new Symfony\Component\Routing\Exception\NoConfigurationException();\n";
+            $code .= "        }\n";
+        }
```

Comment:
I like this approach, makes a lot of sense to me.

Reply:
👏 ingenious indeed.